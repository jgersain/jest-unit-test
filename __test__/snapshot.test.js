import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe("Instantaneas", () => {
  test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  test('Siempre fallara la instantanea', () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20),
    }
    expect(user).toMatchSnapshot();
  });

  test('Permitir excepciones dentro del cÃ³digo (snapshots)', () => {
    const user = {
      name: "Oscar Barajas",
      id: Math.floor(Math.random() * 20),
    }
    expect(user).toMatchSnapshot({
      id: expect.any(Number)
    });
  })
});
  