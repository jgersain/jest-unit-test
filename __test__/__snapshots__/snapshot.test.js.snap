// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instantaneas Permitir excepciones dentro del c√≥digo (snapshots) 1`] = `
Object {
  "id": Any<Number>,
  "name": "Oscar Barajas",
}
`;

exports[`Instantaneas Siempre fallara la instantanea 1`] = `
Object {
  "createAt": 2019-09-24T05:06:00.083Z,
  "id": 8,
}
`;

exports[`Instantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;
